### PR TITLE
hack to get vlc udp media streaming to work on Windows

### PR DESCRIPTION
**Pull request by arBmind**
Some changes that should be properly included to support reading video streams with VLC into obs-studio. Right now it's hacked to work for me. (Time runs out too fast)

### CODE REVIEW

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

File: plugins/vlc-video/vlc-video-source.c [hunk 1]
```diff
@@ -399,6 +399,8 @@ static void add_file(struct vlc_source *c, struct darray *array,
 		new_media = create_media_from_file(path);
 
 	if (new_media) {
+		libvlc_media_add_option_(new_media, ":network-caching=100");
```
Comment (arBmind) [inline plugins/vlc-video/vlc-video-source.c]: This should really be an option. 100ms is to reduce buffering on a local network device.

File: plugins/vlc-video/vlc-video-source.c [hunk 2]
```diff
@@ -387,7 +387,7 @@ static void add_file(struct vlc_source *c, struct darray *array,
 
 	dstr_copy(&new_path, path);
 #ifdef _WIN32
-	dstr_replace(&new_path, "/", "\\");
+	//dstr_replace(&new_path, "/", "\\");
```
Comment (arBmind) [inline plugins/vlc-video/vlc-video-source.c]: slash conversion should not be done to urls.

File: plugins/vlc-video/vlc-video-source.c [hunk 3]
```diff
@@ -303,7 +303,7 @@ static unsigned vlcs_video_format(void **p_data, char *chroma, unsigned *width,
 
 	new_format = convert_vlc_video_format(chroma, &new_range);
 
-	libvlc_video_get_size_(c->media_player, 0, width, height);
+	//int res = libvlc_video_get_size_(c->media_player, 0, width, height);
```
Comment (arBmind) [inline plugins/vlc-video/vlc-video-source.c]: For streams this will not give the right frame size.
The arguments are already set, why is this needed anyways?

File: plugins/vlc-video/vlc-video-source.c [hunk 4]
```diff
@@ -76,7 +76,7 @@ static libvlc_media_t *get_media(struct darray *array, const char *path)
 
 static inline libvlc_media_t *create_media_from_file(const char *file)
 {
-	return libvlc_media_new_path_(libvlc, file);
+	return libvlc_media_new_location_(libvlc, file);
```
Comment (arBmind) [inline plugins/vlc-video/vlc-video-source.c]: for urls a different call is required.

File: plugins/vlc-video/vlc-video-plugin.h [hunk 5]
```diff
@@ -1,16 +1,16 @@
 #include <obs-module.h>
-#include <libvlc.h>
+#include <vlc/libvlc.h>
```
Comment (arBmind) [inline plugins/vlc-video/vlc-video-plugin.h]: for we internally libvlc.h includes stuff with a vlc/ subfolder. I could not find a way to get this working without changing the includes here as well.

Comment (jp9000) [inline plugins/vlc-video/vlc-video-plugin.h]: This definitely is not necessary.  I've tested this on all three major operating systems and this has never been a problem.  In the cmake module in cmake/Modules/FindLibVLC.cmake, you'll notice it searches suffixes: vlc, include/libvlc.  This is most likely a problem with your setup rather than with the code itself.

**Comment by jp9000** [comment]
Going to make a few modifications then merge:

1.) The commit message needs to be prefixed with the module.  This one for example would be prefixed with "vlc-video: ".  The title should be something like "vlc-video: Fix URLs not working on windows", and the message should explain the issue in detail.
2.) As stated adding vlc/ to the includes isn't necessary and probably means there's an issue with your configuration.  Make sure you're cloning the VLC repository and setting the VLCPath cmake variable to the repository directory.  That's all you need to make it work.
3.) libvlc_video_get_size was used to correct sizing issues with video files.  It's hard to explain; VLC will use a size different than the video file's actual size, like it'll try to use 1920x1088 instead of 1920x1080, which makes it render incorrectly in the program (I assume it does that by default for optimization reasons).  It should always be used for video files, but doesn't work with URLs; it just returns 0x0.  That's my fault.  I'll fix it so it works, and I'll make sure to add a comment for this.
4.) libvlc_media_new_location shouldn't be used for files.  I'll make it detect whether it was a URL by seeing if there's a "://" in the string, indicating a URL.

File: plugins/vlc-video/vlc-video-plugin.h (hunk 5 above)
Comment (arBmind) [inline plugins/vlc-video/vlc-video-plugin.h]: Can you add instructions, what a developer should download and use to get VLC support compiled?

I downloaded the [7z archive](https://download.videolan.org/pub/videolan/vlc/2.2.4/win64/vlc-2.2.4-win64.7z) and used the sdk folder. But I could not find any way to get it working without changing the includes.

Comment (Gol-D-Ace) [inline plugins/vlc-video/vlc-video-plugin.h]: Clone the VLC Git Repo and point cmake to the folder.
